portfolio/src/main/webapp/comments.js

```diff
@@ -0,0 +1,153 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// TODO: Rewrite with a general class to keep track of things more easily...
+
+/**
+ * Clear the message div
+ */
+function clearMessageDiv() {
+  document.getElementById("message-container").innerHTML = "";
+}
+
+
+/**
+ * Return the TOTAL number of undeleted comments currently in datastore 
+ */
+function numComments() {
+  (fetch('/num-comments').then(response => response.text()).then(num => {
```

Comment:
what happens if this throws an error

Reply:
Added basic error handling; will include something more robust in future commits.